Unnamed Roguelike Game/Player.gd

```diff
@@ -4,9 +4,26 @@ var player_speed = 300.0
 
 @onready var player_health = 100
 
+var enemies_in_hitbox = []
+
 func _physics_process(delta):
 
 	var input_direction = Input.get_vector("left", "right", "up", "down").normalized()
 	velocity = input_direction * player_speed * delta
 	
 	move_and_collide(velocity)
+
+
+func _on_area_2d_body_entered(body):
+	if "Enemy" in body.name:
+		player_health -= 10
+		enemies_in_hitbox.push_back(body.name)
+		if player_health == 0:
+			self.queue_free()
+		#print(enemies_in_hitbox)
+
+
+func _on_area_2d_body_exited(body):
+	if "Enemy" in body.name:
+		enemies_in_hitbox.pop_back()
+		#print(enemies_in_hitbox)
```

Comment:
This is all good and I like how you solved the null reference to Player issue, but this doesn't currently check if the enemies are still touching the player, right? When I was testing this on your branch, if an enemy was touching me, they would deal damage just one time, regardless of if they are still touching the player, which is not good. I think I mentioned some ways we could probably solve this issue in the discord.

You could set up a Timer for an invulnerability check, where the timer on a player starts when it takes damage, and once that timer is up, you set up the timer ended signal to check if enemies_in_hitbox is not empty, and if it isn't empty, then the player should be taking damage because an enemy is still touching it.

If that doesn't make sense, wait until I get home and I can hopefully explain better. For the time being though, I'm just going to request changes and you can try to solve the issue or wait until I can provide some assistance!

Reply:
See new commit for details, but I-Frames are now IN